core/meta/src/TClass.cxx

```diff
@@ -5464,29 +5464,24 @@ TClass *TClass::LoadClassCustom(const char *requestedname, Bool_t silent)
 
 void TClass::LoadClassInfo() const
 {
-   R__LOCKGUARD(gInterpreterMutex);
+   // Load class info if not already loaded and auto-parsing is not suspended
+   if (!fClassInfo && !gInterpreter->IsAutoParsingSuspended()) {
```

Comment:
I cannot see how a pointer can be only partially set.

Your second comment does make sense, though. What do I need to avoid when auto parsing is suspended? It's not clear to me. I thought `AutoParse(GetName())` should only be called if it's not suspended...